code/modules/research/experimentor/loot/_loot_definer.dm

```diff
@@ -0,0 +1,222 @@
+/////////////////////////////////////LOOT DEFINER/////////////////////////////////////
+// Holds all the messy stuff to do with name and type picking, loot lists etc.
+// define() returns a named unboxed item based on stats and loot lists.
+// The actual items and their purposes should be defined elsewhere,
+// though they can be initialized here as part of the loot picking.
+//
+// Expansion sections (Searchable):
+// O.   LEAVE THIS PART ALONE
+// I.   LOOT LIST
+// II.  LOOT DEFINITIONS
+// III. SECOND NAME
+// IV.  THIRD NAME
+// V.   DESCRIPTION
+
+
+#define RARITY_COMMON 0
+#define RARITY_UNCOMMON 1
+#define RARITY_RARE 2
+#define RARITY_VERYRARE 3
+
+//////////////////////// O. LEAVE THIS PART ALONE //////////////////////////
+// Unless you have a good idea of how this works.
+// If you're just adding new loot you shouldn't have to touch this.
+
+/obj/experimentor/loot_definer
+    name = "loot definer"
+    desc = "If you can see this, something is very, very wrong."
+    var/stability = 0
+    var/potency = 0
+    var/raritylevel = 0
+    var/box_type
+    var/item_category
+    var/item_type
+    var/obj/loot_item
+
+    // Set to FALSE during item definition to use the default name or description for the new object.
+    var/loot_named
+    var/loot_described
+    // Keywords used to choose names for the second and third parts of the generated name.
+    // When defining new keywords, try to use ones that fit in the auto-generated descriptions.
+    var/list/loot_keywords
+    // Unique Names - If instantiated in the definition, will use one of the contained names instead of generating the third name.
+    var/list/unique_names
+
+// Returns a finished item from the stats supplied to the experimentor.
+/obj/experimentor/loot_definer/proc/define(var/stability_in, var/potency_in, var/base_name, var/rare_level, var/itemcategory)
```

Comment:
should this really be an obj instead of a datum?

Reply:
probably not.

Reply:
Might be better as a datum yeah,